src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py

```diff
@@ -16565,94 +16582,115 @@ def dt_round(
         if nonexistent != "raise":
             ErrorMessage.parameter_not_implemented_error("nonexistent", method_name)
 
-        slice_length, slice_unit = rule_to_snowflake_width_and_slice_unit(
-            rule=freq  # type:  ignore[arg-type]
-        )
-
-        if slice_unit not in SUPPORTED_DT_FLOOR_CEIL_FREQS or slice_unit == "second":
-            ErrorMessage.parameter_not_implemented_error(f"freq={freq}", method_name)
-
-        # We need to implement the algorithm for rounding half to even whenever
-        # the date value is at half point of the slice:
-        # https://en.wikipedia.org/wiki/Rounding#Rounding_half_to_even
-
-        # First, we need to calculate the length of half a slice.
-        # This is straightforward if the length is already even.
-        # If not, we then need to first downlevel the freq to a
-        # lower granularity to ensure that it is even.
+        dtype = self.index_dtypes[0] if include_index else self.dtypes[0]
+        if is_datetime64_any_dtype(dtype):
+            slice_length, slice_unit = rule_to_snowflake_width_and_slice_unit(
+                rule=freq  # type:  ignore[arg-type]
+            )
 
-        def down_level_freq(slice_length: int, slice_unit: str) -> tuple[int, str]:
-            if slice_unit == "minute":
-                slice_length *= 60
-                slice_unit = "second"
-            elif slice_unit == "hour":
-                slice_length *= 60
-                slice_unit = "minute"
-            elif slice_unit == "day":
-                slice_length *= 24
-                slice_unit = "hour"
-            else:
-                f"Snowpark pandas 'Series.dt.round' method doesn't support setting 'freq' parameter with '{slice_unit}' unit"
-            return slice_length, slice_unit
+            if (
+                slice_unit not in SUPPORTED_DT_FLOOR_CEIL_FREQS
+                or slice_unit == "second"
+            ):
+                ErrorMessage.parameter_not_implemented_error(
+                    f"freq={freq}", method_name
+                )
 
-        if slice_length % 2 == 1:
-            slice_length, slice_unit = down_level_freq(slice_length, slice_unit)
-        half_slice_length = int(slice_length / 2)
+            # We need to implement the algorithm for rounding half to even whenever
```

Comment:
@sfc-gh-helmeleegy can we use `bround` for this case too?

Reply:
It might work. I didn't try it, but note that `bround` does not take datetime inputs. So will need some pre and post processing if it can indeed work here.
@sfc-gh-nkumar Can we add a TODO to consider this later. I don't think it's high priority give that we already have a working solution.